### PR TITLE
Npc lifecycle

### PR DESCRIPTION
**Pull request by SamuelllG**
Health, Dying and regeneration implemented

### CODE REVIEW

File: Assets/Scripts/Entity/Enemy/EnemyState.cs [hunk 1]
```diff
@@ -2,17 +2,81 @@
...
+            health = updatedValue;
+        }
+
+        private void Regenerate()
+        {
+            if (_probability.GetProbability(regenerationProbability))
+            {
+                System.Random random = new System.Random();
+                float x = (float) random.NextDouble();
+                Debug.Log("x: " + x);
```
Comment (DavidPgl) [inline Assets/Scripts/Entity/Enemy/EnemyState.cs]: If not needed, remove debugs

**Review by DavidPgl — CHANGES_REQUESTED** [review]
No summaries?:(

**Comment by SamuelllG** [comment]
SORRY, I forgot to do a lookover... will add descriptions, etc. !

**Review by DavidPgl — APPROVED** [review]

**Review by thuemmels — APPROVED** [review]
I think using probabilities is a bit weird when it comes to regeneration, but it works, so why not :D

**Comment by SamuelllG** [comment]
I used probabilities so that the enemy doesn't regenerate consistently

**Pull request merged**